llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -198,13 +231,15 @@ static bool hasUndefinedMergeOp(const MachineInstr &MI,
   if (UseMO.getReg().isPhysical())
     return false;
 
-  if (MachineInstr *UseMI = MRI.getVRegDef(UseMO.getReg())) {
+  if (const MachineInstr *UseMI =
```

Comment:
I don't think it's safe to use getVRegDef to find the register definition here. Even though it's always a virtual register, it's still possible for it to be defined multiple times in non-SSA environment.

Reply:
Sorry, this was written when we still had the pre-RA code path. Now that the option is gone we can replace this entire function with

```c++
/// Return true if the inactive elements in the result are entirely undefined.
/// Note that this is different from "agnostic" as defined by the vector
/// specification.  Agnostic requires each lane to either be undisturbed, or
/// take the value -1; no other value is allowed.
static bool hasUndefinedMergeOp(const MachineInstr &MI,
                                const MachineRegisterInfo &MRI,
                                const LiveIntervals *LIS) {

  unsigned UseOpIdx;
  if (!MI.isRegTiedToUseOperand(0, &UseOpIdx))
    // If there is no passthrough operand, then the pass through
    // lanes are undefined.
    return true;
  return MI.getOperand(UseOpIdx).isUndef();
}
```

Reply:
To clarify, the post-ra LIS path never would have gone past the `if (UseMO.getReg().isPhysical())` statement, because all the vector registers would have been allocated at that point